src/expressions/visitors/EvalVisitor.cpp

```diff
@@ -164,4 +176,27 @@ EvaluationResult::EvaluationResult(const std::variant<double, std::vector<double
 {
 }
 
+EvaluationResult EvaluationResult::shiftResult(int shiftValue) const
+{
+    return EvaluationResult(
+      std::visit([&shiftValue](const auto& l) -> std::variant<double, std::vector<double>>
+                 { return shift(l, shiftValue); },
+                 value_));
```

Comment:
Can you explain why we need to handle double & vectors here ?

Reply:
because (1+p)[t-1] where p is constant parameter, shift has no effect on this that's why the input is returned in shift(doubl)